lib/autobahn/consumer.rb

```diff
@@ -121,10 +120,17 @@ def setup!
     end
 
     def deliver(handler, timeout)
-      begin
+      while @deliver.get
         headers, message = self.next(timeout)
         handler.call(headers, message) if headers
-      end while @deliver.get
+      end
+      while (headers, message = self.next(0))
+        if headers
+          handler.call(headers, message)
+        else
+          break
+        end
+      end
```

Comment:
It turns out that the `handler.call(*entry)` is about 60% slower than `handler.call(headers, message)` in my version of JRuby, so that's a reason for staying with that part of the verbosity at least.